src/fe-text/irssi.c

```diff
@@ -331,6 +342,11 @@ int main(int argc, char **argv)
 	/* Does the same as g_main_run(main_loop), except we
 	   can call our dirty-checker after each iteration */
 	while (!quitting) {
+		if (sigterm_received) {
+			signal_emit("gui exit", 0);
+			quitting = TRUE;
```

Comment:
This is redundant (which is why the `quit_on_hup` path doesn't do it)

Reply:
Thanks for the feedback and this has been addressed in the new proposal.